arch/Kconfig

```diff
@@ -502,7 +502,8 @@ config ARCH_HAVE_PERF_EVENTS
 
 config ARCH_PERF_EVENTS
 	bool "Configure hardware performance counting"
-	default y
+	default y if SCHED_CRITMONITOR || SCHED_IRQMONITOR || RPTUN_PING || SEGGER_SYSVIEW
```

Comment:
@xiaoxiang781216 what about this ? I got rid of the `ARCH_NEED_PERF` flag but in that case we need long `default y` condition for `ARCH_PERF_EVENTS`

Reply:
after #10384, all perf component doesn't depend on ARCH_PERF_EVENTS anymore, so I think we can simply change the default value to n without if statement.

Reply:
If we set the default ARCH_PERF_EVENTS to n then the default source for perf will be systime which usually has poor resolution.  In this case, some of the debugging functions will be useless (eg. systemview). I think this may cause problems for users. That's why I think ARCH_PERF_EVENTS=y should be y when some debug features are enabled